app/models/submission.rb

```diff
@@ -30,6 +30,27 @@ def self.create_by_timestamp(grouping, timestamp)
      repo.close
      submission
   end
+  
+  def apply_penalty_or_add_grace_credits(grouping,
```

Comment:
Does this function need to be duplicated in both submission.rb and submissions_job.rb?